infra/testApi.mjs

```diff
@@ -3,177 +3,317 @@ import { strict as assert } from 'node:assert';  // use strict equality everywhe
+`
 
-// Test with a likely missing job-id
-const badTimelineRSP = await fetch(makeEndpoint("/timeline/42069"), { method: 'GET' })
-assert.equal(badTimelineRSP.status, 404)
-const check_missing_job = await fetch(makeEndpoint(`/check-status/42069`), { method: 'GET' })
-assert.equal(check_missing_job.status, 202)
+await simulateOdysseyCall(FPCoreFormula);
+await simulateOdysseyCall(FPCoreFormula2);
+await simulateOdysseyCall(FPCoreFormula3);
+await simulateOdysseyCall(det_44);
 
-// improve-start endpoint
-const URIencodedBody = "formula=" + encodeURIComponent(FPCoreFormula)
-const startResponse = await fetch(makeEndpoint("/api/start/improve"), {
-  method: 'POST',
-  headers: {
-    'Content-Type': 'application/x-www-form-urlencoded',
-  },
-  body: URIencodedBody
-})
-const testResult = (startResponse.status == 201 || startResponse.status == 202)
-assert.equal(testResult, true)
-const improveResultPath = startResponse.headers.get("location")
-let counter = 0
-let cap = 100
-// Check status endpoint
-let checkStatus = await fetch(makeEndpoint(improveResultPath), { method: 'GET' })
-/*
-This is testing if the /api/start/improve test at the beginning has been completed. The cap and counter is a sort of timeout for the test. Ends up being 10 seconds max.
-*/
-while (checkStatus.status != 201 && counter < cap) {
-  counter += 1
-  checkStatus = await fetch(makeEndpoint(improveResultPath), { method: 'GET' })
-  await new Promise(r => setTimeout(r, 100)); // ms
-}
-assert.equal(checkStatus.statusText, 'Job complete')
+// MARK: Analyze
+await callAnalyze(FPCoreFormula, [[[
+  14.97651307489794
+], 0.12711304680349078]], [[[14.97651307489794], "2.3"]], false);
 
-// up endpoint
-const up = await fetch(makeEndpoint("/up"), { method: 'GET' })
-assert.equal('Up', up.statusText)
-// TODO how do I test down state?
+await callAnalyze(FPCoreFormula, [[[
+  14.97651307489794
+], 0.12711304680349078]], [[[14.97651307489794], "2.3"]], true);
 
-// Sample endpoint
-const sampleBody = {
-  method: 'POST',
-  body: JSON.stringify({ formula: FPCoreFormula2, seed: 5 })
+async function callAnalyze(fpcore, p_context, result_huh, async_huh) {
+  console.debug(`Analyze for ${fpcore}`);
+  const errors_body = {
+    method: 'POST', body: JSON.stringify({
+      formula: fpcore, sample: p_context
+    })
+  };
+  var alt_error = null;
+  if (async_huh) {
+    const analyze_rsp = await callAsyncAndWaitResult("/api/start/analyze", errors_body, CHECK_CORS);
+    assert.equal(analyze_rsp.status, 200);
+    alt_error = await analyze_rsp.json();
+  } else {
+    const analyze_rsp = await fetchAndCheckRSPHeaders("/api/analyze", errors_body, CHECK_CORS);
+    assert.equal(analyze_rsp.status, 200);
+    alt_error = await analyze_rsp.json();
+  }
+  assert.equal(Object.values(alt_error).length, 4);
+  assertIdAndPath(alt_error);
+  assert.equal(alt_error.points.length == p_context.length, true);
+  if (result_huh != null) {
+    assert.deepEqual(alt_error.points, result_huh);
+  }
```

Comment:
Why does the caller have to pass in the expected result? Why not write the `assert.deepEqual` in the caller?